File: ArrayParition.java
//Time Complexity - O(nlogn)
class Solution {
    public int arrayPairSum(int[] nums) {
        if(nums == null | nums.length == 0){
            return 0;
        }
        Arrays.sort(nums);
        int sum = 0;
        for(int i = 0; i < nums.length; i+= 2){
            sum = sum + nums[i]; 
        }
        return sum;
    }
}

File: arrayParition.py
#Time: O(nlogn)
#Space: O(1)

def findMax(nums):
    if nums is None or len(nums) == 0:
        return None
    
    nums.sort()
    i = 0
    result = 0
    while (i < len(nums)):
        result += min(nums[i], nums[i+1])
        i += 2
    
    return result
